docs/testing.md

```diff
@@ -39,20 +39,54 @@ They generally follow TPCH but are not guaranteed to be the same.
 You can run the unit tests against different versions of Spark using the different profiles. The
-default version runs again Spark 3.0.0, to run against other version use one of the following profiles:
-   - `-Pspark301tests` (Spark 3.0.1)
-   - `-Pspark302tests` (Spark 3.0.2)
-   - `-Pspark310tests` (Spark 3.1.0)
+default version runs against Spark 3.0.0, to run against other versions use one of the following
+ profiles:
+   - `-Pspark301tests` (spark 3.0.1)
+   - `-Pspark302tests` (spark 3.0.2)
+   - `-Pspark310tests` (spark 3.1.0)
+
+Please refer to the [pom.xml](tests/pom.xml) to see the list of profiles supported
+apache spark specific configurations can be passed in by setting environment-variable spark_conf 
+
+Examples: 
+-To run tests against Apache Spark 3.1.0, 
+ `mvn -P spark310tests test -DwildcardSuites="com.nvidia.spark.rapids.ParquetWriterSuite"`.
+- To pass Apache Spark configs `--conf spark.dynamicAllocation.enabled=false --conf spark.task.cpus=1` do something like.
+ `SPARK_CONF="spark.dynamicAllocation.enabled=false,spark.task.cpus=1" mvn ...`
+- To run test ParquetWriterSuite in package com.nvidia.spark.rapids, issue `mvn test -DwildcardSuites="com.nvidia.spark.rapids.ParquetWriterSuite"`
 
 ## Integration tests
 
-Integration tests are stored in the [integration_tests](../integration_tests/README.md) directory.
-There are two frameworks used for testing. One is based off of pytest and pyspark in the 
-`src/main/python` directory. These tests will run as a part of the maven build if you have the environment
-variable `SPARK_HOME` set.
+Integration tests are stored in the [integration_tests](../integration_tests) directory. 
+There are two frameworks used for testing. one is based off of pytest and pyspark in the 
+`src/main/python` directory. These tests will run as a part of the maven build if you have the 
+environment variable `SPARK_HOME` set.
+
+The suggested way to run these tests is to use the shell-script file located in the module folder 
+called [run_pyspark_from_build.sh](integration_tests/run_pyspark_from_build.sh). This script takes 
+care of some of the flags that are required to run the tests which will have to be set for the 
+plugin to work. it will be very useful to read the contents of the 
+[run_pyspark_from_build.sh](integration_tests/run_pyspark_from_build.sh) to get a better insight 
+into what is needed as we constantly keep working on to improve and expand the plugin-support.
+
+The tests are written python and run with pytest and the script honors pytest parameters. Some handy flags are:
+- `-k` <pytest-file-name>. This will run all the tests in that test file.
+- `-k` <test-name>. This will also run an individual test.
+- `-s` Doesn't capture the output and instead prints to the screen.
+- `-v` Increase the verbosity of the tests
+- `-rfexxs` Show extra test summary info as specified by chars: (f)ailed, (e)rror, (x)failed, (x)passed, (s)kipped
```

Comment:
These flags are wrong.  The first clue is the same character meaning two different options.  It should be something like this:
```suggestion
- `-r fExXs` Show extra test summary info as specified by chars: (f)ailed, (E)rror, (x)failed, (X)passed, (s)kipped
```